xcb/xml/auth.scm

```diff
@@ -75,10 +75,11 @@
     `((display . ,(match:substring m 1))
       (screen . ,(match:substring m 2))))
 
-  (define (display-match? xauth-display display-string)
-    (string=
-     (assq-ref (parse-display-name display-string) 'display)
-     xauth-display))
+(define (display-match? xauth-display display-string)
+  (or (string=
```

Comment:
I would swap string= and string-null? just to avoid an extra evaluation. It seems to me that an empty display is a more common case.

Reply:
Yeah I think that's an all-around good idea.